lib/alexa_request_verifier/cert_cache.ex

```diff
@@ -0,0 +1,66 @@
+defmodule AlexaRequestVerifier.CertCache do
+  @moduledoc """
+  This module caches successfully verified Amazon certificates.
+  """
+
+  use GenServer
+
+  require Logger
+
+  @server __MODULE__
+  @table :alexa_request_verifier_certs
+
+  # client
+
+  @doc """
+  Return either the last successfully verified Amazon certificate,
+  or nil if this is the first time the cache has been accessed.
+  """
+  @spec get_cert(cert_url :: String.t()) :: String.t() | nil
+  def get_cert(cert_url) do
+    # check the cache before calling the server
+    case :ets.lookup(@table, cert_url) do
+      [{_url, cert}] -> cert
+      [] -> GenServer.call(@server, {:fetch, cert_url})
```

Comment:
Since you have a separate store call, you shouldn't need to call the server at all here.